### DISCUSSION TITLE
Limits for [all] and [prl] nodes

### DISCUSSION THREAD

**Discussion by Inferdy** [mxgmn/MarkovJunior#32] [Ideas]
It would be cool if we could limit the amount of `all` and `prl` nodes by number or percent of their current number.

**Comment by mxgmn** [comment]
I don't understand what you mean by "limit by percent of their current number", but you can limit the amount of times `all` and `prl` nodes are applied with the `steps` attribute.

**Reply by Inferdy** [reply]
I mean the number of convertations during one pass. We convert all the matches in case of [prl] right now, I wish we could set the maximum number of convertations per pass. It's like [one] with step attribute, but it's actually different.

**Reply by mxgmn** [reply]
I see. You can also use the `p` attribute in `prl` nodes to set the probability of a rule being applied. Example: in [Apartemazements](https://github.com/mxgmn/MarkovJunior/blob/main/models/Apartemazements.xml) only 25% of ceiling locations are converted into light sources.

**Reply by Inferdy** [reply]
But how does probability work? Does it "roll a dice" for each match or select random set of matches big enough to satisfy predefined proportion?

**Reply by mxgmn** [reply]
It rolls a dice for each match, yes.
